xenforo/library/bdApi/ControllerApi/Notification.php

```diff
@@ -103,10 +103,23 @@ public function actionPostCustom()
     public function actionPostRead()
     {
         $this->_assertRegistrationRequired();
-        $visitor = XenForo_Visitor::getInstance();
 
-        if ($visitor['alerts_unread'] > 0) {
-            $this->_getAlertModel()->markAllAlertsReadForUser($visitor['user_id']);
+        if ($this->_input->inRequest('notification_id')) {
+            $alertId = $this->_input->filterSingle('notification_id', XenForo_Input::UINT);
+            $alert = $this->_getAlertOrError($alertId);
+
+            if (empty($alert['view_date'])) {
+                $markAlertRead = array($this->_getAlertModel(), 'bdAlerts_markAlertRead');
+                if (is_callable($markAlertRead)) {
+                    call_user_func($markAlertRead, $alert);
+                }
```

Comment:
Nếu api mark as read đích danh một alert mà hệ thống không hỗ trợ thì trả về 404.

Reply:
Ủa noPermission chứ sao lại notFound a?

Reply:
Đáng ra phải là not supported nhưng không có mã đó nên thôi thì 403 đi =))